packages/swap/src/routes/v2/utils.ts

```diff
@@ -495,17 +500,19 @@ export const getCcmParams = (
     : undefined;
 };
 
-export const getRolledSwapsInitialData = (
-  swapDepositChannel: SwapChannelData | null | undefined,
-) => ({
-  swappedOutputAmount: new Prisma.Decimal(0),
-  swappedIntermediateAmount: new Prisma.Decimal(0),
-  swappedInputAmount: new Prisma.Decimal(0),
-  executedChunks: 0,
-  currentChunk: null as null | NonNullable<SwapRequestData['swaps']>[number],
-  lastExecutedChunk: null as null | NonNullable<SwapRequestData['swaps']>[number],
-  isDca: Boolean(
-    swapDepositChannel?.dcaChunkIntervalBlocks && swapDepositChannel.dcaChunkIntervalBlocks > 1,
-  ),
-  fees: [] as SwapFee[],
-});
+export const getRolledSwapsInitialData = (swapRequest: SwapRequestData | undefined | null) => {
+  const isDca =
+    swapRequest?.dcaNumberOfChunks != null
+      ? swapRequest?.dcaNumberOfChunks > 1
+      : (swapRequest?.swapDepositChannel?.dcaChunkIntervalBlocks ?? 0) > 1;
```

Comment:
can simplify this a little
```suggestion
  const isDca = (swapRequest?.dcaNumberOfChunks ?? 1) > 1;
```